lib/Serialization/SerializeDoc.cpp

```diff
@@ -376,6 +378,16 @@ static void writeDeclCommentTable(
         // public+internal when testing enabled).
         if (VD->getEffectiveAccess() < swift::AccessLevel::Public)
           return false;
+
+        // If a filter is specified, we should avoid serializing doc-comments
+        // of those matching decls.
+        if (!DocCommentFilter.empty()) {
+          auto Name = VD->getBaseName();
+          if (!Name.isSpecial() &&
+              Name.getIdentifier().str().startswith(DocCommentFilter)) {
```

Comment:
Do initializers and subscripts get their special case here too? (where we look at the first parameter instead)

Reply:
The cases we've seen don't include initializers and subscripts.

Reply:
Yes, but we don't want to have to do *another* submission of the compiler when someone asks "how do I do it for initializers and subscripts?"

Reply:
That's fair. I have updated the PR to check if the first label of subscripts and initializers matches the given filter; also the logic has been refactored to another function `shouldIncludeDecl`.